### ISSUE TITLE
File encoding is improperly selected when using "Open with Code" from context menu

### ISSUE THREAD

**Issue by FallenLeaf54** [microsoft/vscode#122632]
- VS Code Version: 1.55.2 (user setup)
- OS Version: Windows_NT x64 10.0.19042

Steps to Reproduce:

1. Create a sample text file with an extension indicating a programming language which prefers some non-Unicode encoding (in my case it's a BAT file with encoding `cp852`, which I will be using here as an example).
2. Configure VS Code to use this encoding every time a BAT file is opened by placing `"[bat]": { "files.encoding": "cp852" }` inside `settings.json`.
3. __Open VS Code first__ and then open the file created in Step 1 in any way, including File -> Open or by drag-and-drop or by selecting _Open with Code_ from context menu. File is then opened and encoding properly selected according to preference given in Step 2.
4. Now close VS Code and instead open the file using _Open with Code_. Encoding given in Step 2 is not taken into consideration and `UTF-8` is improperly selected which displays gibberish if the file contained non-ASCII characters.

If you complete Step 2, you can test this bug yourself on this [test_file.zip](https://github.com/microsoft/vscode/files/6397253/test_file.zip).

Does this issue occur when all extensions are disabled?: Yes

**Comment by FallenLeaf54** [comment]
@bpasero So basically in November it's going to be **four** years since the issue has been known, you confirmed the bug yourself in the linked 2017 bug report.
What's more, it has been reported by multiple users time and time again over this time and each thread was closed down by bots, marked duplicate so other developers have no chance of seeing (who's going to comb through closed duplicate threads?).
In short, the bug happily continues to exist.
